### PR TITLE
CI/CD for jaseci

### PR DESCRIPTION
**Pull request by ashagraw91**
This PR starts the introduction of CI/CD for jaseci open source github repo, using github actions.
* Adds CI jobs for jaseci_core, includes unit tests and linting
* Adds a WIP CI job skeleton for jaseci_serv, no tests running as part of that just yet.
* Adds a WIP CD job for pushing new python package to PyPi. Needs to configure credentials to enable this.

### CODE REVIEW

File: .github/workflows/Jaseci_core_build.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install black flake8==3.7.9 isort
+        
+    - name: Install Jaseci
+      run: |
+        source jaseci_core/install.sh
```
Comment (ypkang) [inline .github/workflows/Jaseci_core_build.yml]: lets directly run the pip install command here instead of using bash scripts.

File: .github/workflows/Jaseci_core_build.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    - name: check Jaseci
+      run: |
+        jsctl
+    - name: Test Jaseci Setup
+      run: |
+        source test.sh
```
Comment (ypkang) [inline .github/workflows/Jaseci_core_build.yml]: similar here, lets use the python unit test command directly.

Comment (ypkang) [reply .github/workflows/Jaseci_core_build.yml]: And lets separate the unit test and flake8 linting to be two jobs

File: python-app.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
```
Comment (ypkang) [inline python-app.yml]: remove this.

File: jaseci_core/install.sh [hunk 4]
```diff
@@ -1 +1 @@
-pip3 install .
+pip3 install jaseci_core/
```
Comment (ypkang) [inline jaseci_core/install.sh]: revert this change.

File: jaseci_serv/install.sh [hunk 5]
```diff
@@ -1 +1 @@
-pip3 install .
+pip3 install jaseci_serv/
```
Comment (ypkang) [inline jaseci_serv/install.sh]: revert this change.

File: .github/workflows/python-publish.yml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+        python-version: '3.x'
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install setuptools wheel twine
+        pip install build
+        
+    - name: Build and Publish
+      env: 
+        TWINE_USERNAME: ${{ secrets.BHARAT_PYPI_USERNAME }}
```
Comment (ypkang) [inline .github/workflows/python-publish.yml]: Change this to be JASECI_PYPI_USERNAME

File: .github/workflows/python-publish.yml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+        TWINE_PASSWORD: ${{ secrets.BHARAT_PYPI_PASSWORD }}
```
Comment (ypkang) [inline .github/workflows/python-publish.yml]: Change this to be JASECI_PYPI_PASSWORD

File: .github/workflows/Jaseci_serv_build.yml [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        
+    - name: Install Jaseci
+      run: |
+        source jaseci_serv/install.sh
```
Comment (ypkang) [inline .github/workflows/Jaseci_serv_build.yml]: similarly here, lets directly call the install command.

**Review by ypkang — CHANGES_REQUESTED** [review]

**Comment by ypkang** [comment]
* @ashagraw91 to make the changes suggested above
* @bharat-saini-bcs to sign the CLA
* And we can merge this.

**Comment by ypkang** [comment]
and note @ypkang to remember to squash this when merging.

File: .github/workflows/Jaseci_serv_build.yml (hunk 8 above)
Comment (ashagraw91) [reply .github/workflows/Jaseci_serv_build.yml]: this is done

File: .github/workflows/python-publish.yml (hunk 7 above)
Comment (ashagraw91) [reply .github/workflows/python-publish.yml]: this is done

File: .github/workflows/python-publish.yml (hunk 6 above)
Comment (ashagraw91) [reply .github/workflows/python-publish.yml]: done

File: jaseci_serv/install.sh (hunk 5 above)
Comment (ashagraw91) [reply jaseci_serv/install.sh]: done

File: jaseci_core/install.sh (hunk 4 above)
Comment (ashagraw91) [reply jaseci_core/install.sh]: done

File: .github/workflows/Jaseci_core_build.yml (hunk 2 above)
Comment (ashagraw91) [reply .github/workflows/Jaseci_core_build.yml]: done

File: .github/workflows/Jaseci_core_build.yml (hunk 1 above)
Comment (ashagraw91) [reply .github/workflows/Jaseci_core_build.yml]: done

**Review by ypkang — APPROVED** [review]
